compute/cker/include/cker/eigen/redux_functor.h

```diff
@@ -0,0 +1,220 @@
+
+      // Do not create more blocks than there are threads in a pool.
+      const Eigen::Index num_blocks = std::min(max_num_blocks, num_threads);
+
+      // Block size along the outer dimension.
+      const Eigen::Index outer_block_size = Eigen::divup(outer_dim, num_blocks);
+
+      const InputT *input_data = input.template flat<InputT>().data();
+
+      // Allocate temporary buffer for partial reductions.
+      std::vector<AccumT> buffer(num_blocks * inner_dim);
+      AccumT *buffer_data = buffer.data();
+
+      using Buffer =
+        Eigen::TensorMap<Eigen::Tensor<AccumT, 1, Eigen::RowMajor, Eigen::Index>, Eigen::Unaligned>;
+
+      using Input = Eigen::TensorMap<Eigen::Tensor<const InputT, 1, Eigen::RowMajor, Eigen::Index>,
+                                     Eigen::Unaligned>;
+
+      const auto compute = [inner_dim, outer_block_size, buffer_data, input_data,
+                            outer_dim](Eigen::Index start, Eigen::Index limit) -> void {
+        Eigen::Index outer_dim_start = start * outer_block_size;
+        Eigen::Index outer_dim_limit = limit * outer_block_size;
+        outer_dim_limit = std::min(outer_dim, outer_dim_limit);
+
+        Buffer buf(buffer_data + start * inner_dim, inner_dim);
+        for (Eigen::Index i = outer_dim_start; i < outer_dim_limit; ++i)
+        {
+          auto in = Input(input_data + i * inner_dim, inner_dim);
+          auto cast = in.template cast<AccumT>();
+          buf =
+            Eigen::TensorCwiseBinaryOp<BinaryFunctor, const decltype(buf), const decltype(cast)>(
+              buf, cast);
+        }
+      };
+
+      // Compute cost of reducing a single block.
+      const Eigen::Index compute_size = outer_block_size * inner_dim;
+      const Eigen::Index compute_input_bytes = compute_size * sizeof(InputT);
+      const Eigen::TensorOpCost cost(compute_input_bytes,
+                                     0, // We'll be mostly writing to L1, assume store cost is 0
+                                     compute_size *
+                                       Eigen::internal::functor_traits<BinaryFunctor>::Cost);
+
+      device.parallelFor(num_blocks, cost, compute);
+
+      // Aggregate partial results from temporary buffer into first block.
+      auto buf0 = Buffer(buffer_data, inner_dim);
+      // Just sum the buffer up, as inner dimensions is not large in this case.
+      for (int i = 1; i < num_blocks; ++i)
+      {
+        auto buf = Buffer(buffer_data + i * inner_dim, inner_dim);
+        buf0 = Eigen::TensorCwiseBinaryOp<BinaryFunctor, const decltype(buf0), const decltype(buf)>(
+          buf0, buf);
+      }
+      // Write final result to the output.
+      output->template flat<OutputT>() = buf0.template cast<OutputT>().reshape(output_dims);
+    }
+  }
+};
+
+void biasReductionHelper(float *input_backprop_buffer, const Shape &input_backprop_shape,
+                         float *bias_grad_buffer, const Shape &bias_grad_shape)
```

Comment:
This is from @YongseopKim 's [suggestion](https://github.com/Samsung/ONE/pull/12673#issuecomment-1963485619).